include/zephyr/net/coap.h

```diff
@@ -222,6 +223,22 @@ enum coap_content_format {
 	COAP_CONTENT_FORMAT_APP_CBOR = 60               /**< application/cbor */
 };
 
+/**
+ * @brief Set of No-Response option values for CoAP.
+ *
+ * To be used when encoding or decoding a No-Response option defined
+ * in RFC 7967.
+ */
+enum coap_no_response {
+	COAP_NO_RESPONSE_SUPPRESS_2_XX = 0x02,
+	COAP_NO_RESPONSE_SUPPRESS_4_XX = 0x08,
+	COAP_NO_RESPONSE_SUPPRESS_5_XX = 0x10,
```

Comment:
Just a suggestion, please ignore if it does not make sense, but it might be clearer to use the `BIT()` macro here.

Reply:
I was thinking about this, but then it would be:
```
COAP_NO_RESPONSE_SUPPRESS_2_XX = BIT(2-1),
COAP_NO_RESPONSE_SUPPRESS_4_XX = BIT(4-1),
COAP_NO_RESPONSE_SUPPRESS_5_XX = BIT(5-1),
```

So I was considering
```
#define COAP_NO_RESPONSE_SUPPRESS_BIT(x) BIT((x)-1)

COAP_NO_RESPONSE_SUPPRESS_2_XX = COAP_NO_RESPONSE_SUPPRESS_BIT(2),
...
```

But it felt like an overkill and cluttering the enumeration.

I'm fine to change to any of these if you prefer

Reply:
Hmm, I don't quite follow the (x-1) logic here.
Couldn't we just write
```
COAP_NO_RESPONSE_SUPPRESS_2_XX = BIT(1),
COAP_NO_RESPONSE_SUPPRESS_4_XX = BIT(3),
COAP_NO_RESPONSE_SUPPRESS_5_XX = BIT(4),
```
The value of the `BIT(x)` is 2^x, not sure why we would need to have -1 there.
Anyway, I am fine with the original way too, just had a suggestion as the enum value was clearly a bit value.

Reply:
I could do it this way, but it's harder to understand why `2_XX` is bit `1` instead of `2`. It looks like a bug at the first glance